tools/cli/pkg/ers/command/instances.go

```diff
@@ -1,43 +1,141 @@
 package command
 
 import (
-	"fmt"
-	"io/ioutil"
-
+	"github.com/kyma-project/control-plane/tools/cli/pkg/ers"
+	"github.com/kyma-project/control-plane/tools/cli/pkg/ers/client"
+	"github.com/kyma-project/control-plane/tools/cli/pkg/printer"
 	"github.com/spf13/cobra"
 )
 
-func NewInstancesCommand() *cobra.Command {
+var tableColumns = []printer.Column{
+	{
+		Header:    "INSTANCE ID",
+		FieldSpec: "{.Id}",
+	},
+	{
+		Header:    "GLOBALACCOUNT ID",
+		FieldSpec: "{.GlobalAccountId}",
+	},
+	{
+		Header:    "BROKER ID",
+		FieldSpec: "{.BrokerId}",
+	},
+	{
+		Header:    "MIGRATED",
+		FieldSpec: "{.Migrated}",
+	},
+}
+
+type InstanceFetcher interface {
+	GetAllInstances() ([]ers.Instance, error)
+	GetInstanceById(id string) (ers.Instance, error)
+}
+
+type Filters struct {
+	GlobalAccountID string
+	Migrated        bool
+	NotMigrated     bool
+	InstanceID      string
+
+	// TODO: define more filters
+}
+
+type InstancesCommand struct {
+	corbaCmd        *cobra.Command
+	instanceFetcher InstanceFetcher
+	filters         Filters
+	source          string
+}
 
-	cmd := &cobra.Command{
+func (c *InstancesCommand) Run() error {
+	if c.source != "" {
+		c.instanceFetcher = client.NewFileClient(c.source)
+	} else {
+		// todo: use real client to ers
+		c.instanceFetcher = &emptyFetcher{}
+	}
+
+	tp, _ := printer.NewTablePrinter(tableColumns, false)
+
+	if c.filters.InstanceID != "" {
+		instance, err := c.instanceFetcher.GetInstanceById(c.filters.InstanceID)
+		tp.PrintObj(instance)
+		return err
+	}
+	var result []ers.Instance
```

Comment:
Consider to preallocate slice for performance benefits e.g. `result := make([]ers.Instance, 0, len(instances))`

Reply:
To be fixed in the following PRs.